src/components/SocialFeedItem.vue

```diff
@@ -0,0 +1,283 @@
+      default: undefined,
+    },
+    classId: {
+      type: String,
+      default: undefined,
+    },
+    nftId: {
+      type: String,
+      default: undefined,
+    },
+    memo: {
+      type: String,
+      default: undefined,
+    },
+    granterMemo: {
+      type: String,
+      default: undefined,
+    },
+  },
+  data() {
+    return { isFollowLoading: false };
+  },
+  computed: {
+    senderWallet() {
+      switch (this.type) {
+        case EVENT_TYPE.COLLECT:
+        case EVENT_TYPE.PURCHASE:
+          return this.receiverAddress;
+
+        case EVENT_TYPE.SEND:
+        case EVENT_TYPE.PUBLISH:
+        default:
+          return this.senderAddress;
+      }
+    },
+    receiverWallet() {
+      switch (this.type) {
+        case EVENT_TYPE.COLLECT:
+        case EVENT_TYPE.PURCHASE:
+          return this.senderAddress;
+
+        case EVENT_TYPE.SEND:
+        case EVENT_TYPE.PUBLISH:
+        default:
+          return this.receiverAddress;
+      }
+    },
+    senderInfo() {
+      return this.getUserInfoByAddress(this.senderWallet);
+    },
+    receiverInfo() {
+      return this.getUserInfoByAddress(this.receiverWallet);
+    },
+    senderId() {
+      return this.senderInfo?.displayName;
+    },
+    receiverId() {
+      return this.receiverInfo?.displayName;
+    },
+    senderAvatar() {
+      return this.senderInfo?.avatar;
+    },
+    receiverAvatar() {
+      return this.receiverInfo?.avatar;
+    },
+    formatType() {
+      switch (this.type) {
+        case EVENT_TYPE.COLLECT:
+        case EVENT_TYPE.PURCHASE:
+          return this.$t('nft_social_feed_item_type_collected');
+
+        case EVENT_TYPE.SEND:
+          return this.$t('nft_social_feed_item_type_send');
+
+        case EVENT_TYPE.PUBLISH:
+          return this.$t('nft_social_feed_item_type_publish');
+
+        default:
+          return undefined;
+      }
+    },
+    formatTime() {
+      const date = new Date(this.timestamp);
+      const formattedDate = new Intl.DateTimeFormat('en-US', {
+        hour: 'numeric',
+        minute: 'numeric',
+        month: '2-digit',
+        day: '2-digit',
+      }).format(date);
+      return formattedDate;
+    },
+    formatMessage() {
+      switch (this.type) {
+        case EVENT_TYPE.COLLECT:
+        case EVENT_TYPE.PURCHASE:
+          return this.granterMemo;
+
+        case EVENT_TYPE.PUBLISH:
+        case EVENT_TYPE.SEND:
+        default:
+          return this.memo;
+      }
+    },
+    nftTitle() {
+      return this.nftName || this.classId;
+    },
+    shouldShowNftTitle() {
+      return this.type !== EVENT_TYPE.PUBLISH;
+    },
+    shouldShowMessageIcon() {
+      return this.memo || this.granterMemo;
+    },
+    shouldShowItemCard() {
+      return this.type === EVENT_TYPE.PUBLISH;
+    },
+    shouldShowReceiverInfo() {
+      return this.type !== EVENT_TYPE.PUBLISH;
+    },
+    shouldShowFollow() {
+      return !this.walletFollowees.includes(this.senderWallet);
```

Comment:
In what case we will see a sender that we haven't followed?

Reply:
someone sent/collect NFT to my followee